### PR TITLE
CLOUDSTACK-8650: Fix securitygroups ingress FW for protocol any and 0.0.0.0/0

### PR DESCRIPTION
**Pull request by franklouwers**
When using security groups, adding an ingress rule for protocol "any" with source address 0.0.0.0/0, resulted in no action (as the 0.0.0.0/0 entry was stripped from the array of source ips, but unlike icmp/tcp/udp, no special action was set for the handling of the allow_any flag.

This oneliner only removes 0.0.0.0/0 from the list if the protocol isn't any...

### CODE REVIEW

**Comment by franklouwers** [comment]
JIRA ticket: https://issues.apache.org/jira/browse/CLOUDSTACK-8650

**Comment by DaanHoogland** [comment]
@remibergsma @NuxRo &others can you review this?
@franklouwers this is one of the rare cases I think squashing makes sense ;)

**Comment by bhaisaab** [comment]
LGTM, though I've not tested this with a real host

File: scripts/vm/network/security_group.py [hunk 1]
```diff
@@ -860,8 +860,10 @@ def add_network_rules(vm_name, vm_id, vm_ip, signature, seqno, vmMac, rules, vif
                 for ip in ips:
                     execute("iptables -I " + vmchain + " -p icmp --icmp-type " + range + " " + direction + " " + ip + " -j "+ action)
 
-        if allow_any and protocol != 'all':
-            if protocol != 'icmp':
+        if allow_any
```
Comment (resmo) [inline scripts/vm/network/security_group.py]: missing ":"

**Comment by wilderrodrigues** [comment]
@bhaisaab I was going to ask @franklouwers how did he test his changes, since I would like to test them before giving a LGTM.

Actually, I think you were the only one to LGTM it. :)

Cheers,
Wilder

**Comment by franklouwers** [comment]
All,

I'll check the typo (I know how it happend) later this afternoon. Will also provide logs both before (bad behaviour: rule not installed) and after (good thing: rule installed) later today.

**Comment by bhaisaab** [comment]
@wilderrodrigues yeah, as I said I've not tested it. Just had a glance at the code, but good that @resmo pointed out the typo.

**Comment by franklouwers** [comment]
See updated commit to fix the missing : .

See also https://gist.github.com/franklouwers/d5061b4ef50e2b4253fe with logs of what works, what doesn't work, and how this PR makes it work....

**Comment by wilderrodrigues** [comment]
Awesome @franklouwers ! Thanks for the details.

LGTM :+1: Merging...

**Pull request closed without merging**